spiffeid/src/trustdomain.c

```diff
@@ -1,17 +1,21 @@
 #include "spiffeid/src/trustdomain.h"
 
-spiffeid_TrustDomain spiffeid_TrustDomainFromString(string_t str, err_t *err)
+spiffeid_TrustDomain spiffeid_TrustDomainFromString(const char *uri,
+                                                    err_t *err)
 {
     const char spiffe_scheme[] = "spiffe://";
     const size_t spiffe_scheme_len = sizeof spiffe_scheme - 1;
 
+    string_t str = string_new(uri);
+
     if(!string_contains(str, "://")) {
         // inserts spiffe scheme at beginning of the string
         arrinsn(str, 0, spiffe_scheme_len);
         memcpy(str, spiffe_scheme, spiffe_scheme_len);
```

Comment:
this should be removed if we now allocate a new string and handle the string in place
```suggestion

```

Reply:
actually, this one might be correct.